sonarqube/resource_sonarqube_portfolio.go

```diff
@@ -0,0 +1,364 @@
+package sonarqube
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Portfolio used in Portfolio
+type Portfolio struct {
+	Key           string   `json:"key"`
+	Name          string   `json:"name"`
+	Desc          string   `json:"desc"`
+	Qualifier     string   `json:"qualifier"`
+	Visibility    string   `json:"visibility"`
+	SelectionMode string   `json:"selectionMode"`
+	Branch        string   `json:"branch"`
+	Tags          []string `json:"tags"`
+	Regexp        string   `json:"regexp"`
+}
+
+// Returns the resource represented by this file.
+func resourceSonarqubePortfolio() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSonarqubePortfolioCreate,
+		Read:   resourceSonarqubePortfolioRead,
+		Update: resourceSonarqubePortfolioUpdate,
+		Delete: resourceSonarqubePortfolioDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceSonarqubePortfolioImport,
+		},
+
+		// Define the fields of this schema.
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"qualifier": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"visibility": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "public",
+				ForceNew: true, // I cant find an API endpoint for changing this, even though it's possible in the UI
```

Comment:
I dont see an api either. Lets rewrite this comment as a TODO so that someone in the future can add this functionality if it becomes available in the web_api

Reply:
Changed it to a `TODO` comment